### PR TITLE
TASK: Use GH actions for builds (5.3)

### PR DESCRIPTION
**Pull request by albe**
This is a backport of #2273 with outcommented static analysis and adjusted PHP version matrix.

### CODE REVIEW

**Comment by albe** [comment]
This should do now.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  build:
+    if: "!contains(github.event.head_commit.message, '[skip ci]') && !contains(github.event.head_commit.message, '[skip travis]')"
+    name: "PHP ${{ matrix.php-versions }} Test ${{ matrix.static-analysis != 'no' && matrix.static-analysis || '' }} (deps: ${{ matrix.dependencies }})"
+
+    continue-on-error: ${{ matrix.experimental }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        php-versions: ['7.1', '7.2', '7.3', '7.4']
```
Comment (daniellienert) [inline .github/workflows/build.yml:19]: 7.1 was not tested by travis. Do we need it here?

**Review by daniellienert — APPROVED** [review]
Thanks @albe! 
AFAIS the previous travis matrix seems completely covered (+ PHP 7.1).
PHP test code changes look good by reading

File: Neos.Cache/composer.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
         "neos/utility-opcodecache": "*"
     },
     "require-dev": {
-        "mikey179/vfsstream": "~1.6",
+        "mikey179/vfsstream": "^1.6.1",
```
Comment (kdambekalns) [inline Neos.Cache/composer.json:16]: Is this a conscious change (here and elsewhere?) If so, why?

**Review by kdambekalns — COMMENTED** [review]
Could we maybe extract the memcached changes into their own PR? If we ever need to find them again, nobody will look for them in a CI-related change… 😬

Other than that, awesome!

File: .github/workflows/build.yml (hunk 1 above)
Comment (kdambekalns) [reply .github/workflows/build.yml:19]: Good point. According to https://docs.neos.io/cms/installation-development-setup/system-requirements#php-version-matrix we promise only 72+

File: Neos.Cache/composer.json (hunk 2 above)
Comment (albe) [reply Neos.Cache/composer.json:16]: Yes, due to the lowest-dependency test runs. Flow tests won't work with 1.6.0 due to the missing bugfix for `"uninitialized string offset"` (which you reported to vfsstreams btw :))
Lowest dependency tests still don't pass completely, but one issue less at least

File: .github/workflows/build.yml (hunk 1 above)
Comment (albe) [reply .github/workflows/build.yml:19]: According to https://github.com/neos/flow-development-collection/pull/2390/files#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485L10 we did run 7.1 builds on 5.3 and in composer it's also marked as ~7.1 (https://github.com/neos/flow-development-collection/blob/5.3/composer.json#L9), but we can surely strip the build.

File: Neos.Cache/composer.json (hunk 2 above)
Comment (kdambekalns) [reply Neos.Cache/composer.json:16]: Got it, thanks!

File: .github/workflows/build.yml (hunk 1 above)
Comment (kdambekalns) [reply .github/workflows/build.yml:19]: No, then let's leave it in!

**Comment by albe** [comment]
> Could we maybe extract the memcached changes into their own PR? If we ever need to find them again, nobody will look for them in a CI-related change… 😬

There you go: https://github.com/neos/flow-development-collection/pull/2393

**Review by kdambekalns — APPROVED** [review]

**Pull request merged**